src/viewer.js

```diff
@@ -50,18 +49,34 @@ class Viewer {
   /**
    * @param {!Function} showViewer method that shows the viewer.
    * @param {!Function} hideViewer method that hides the viewer.
+   * @param {!Function} isViewerHidden method that determines if viewer is hidden.
    */
-  setViewerShowAndHide(showViewer, hideViewer) {
+  setViewerShowAndHide(showViewer, hideViewer, isViewerHidden) {
     /** @private {!Function} */
     this.showViewer_ = showViewer;
     /** @private {!Function} */
     this.hideViewer_ = hideViewer;
+    /** @private {!Function} */
+    this.isViewerHidden_ = isViewerHidden;
+  }
+
+  /**
+   * @return {boolean} true if the viewer has already been loaded.
+   * @private
+   */
+  isLoaded_() {
+    return !!this.iframe_ && !!this.viewerMessaging_;
   }
 
   /**
    * Attaches the AMP Doc Iframe to the Host Element.
    */
   attach() {
+    if (this.isLoaded_()) {
+      this.history_.pushState(this.ampDocUrl_);
```

Comment:
use case: if you went back & hid the viewer and then clicked on the element that opens the viewer again. It shouldn't push a new state but instead go forward in history.